editor.planx.uk/src/@planx/components/List/Public/Context.tsx

```diff
@@ -84,6 +84,11 @@ export const ListProvider: React.FC<ListProviderProps> = (props) => {
     // Do not allow a new item to be added if there's still an active item
     if (activeIndex !== -1) return setAddItemError(true);
 
+    // Do not allow new item to be added if it will exceed max
+    if (schema.max && formik.values.userData.length === schema.max) {
+      return setMaxError(true);
+    }
```

Comment:
Moving this here feels like a better user experience than just checking this when a user hits "Continue" (and may have added too many items).